### PR TITLE
Handle caching of windspeed override.

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
None

#### Purpose of change
Fire spread tests are still broken after #51212 

#### Describe the solution
Looks like the override is not applied directly, but instead applied via the weather update mechanism.

#### Describe alternatives you've considered
???

### CODE REVIEW

**Comment by kevingranade** [comment]
Ok so it's either not weather or I'm doing the override wrong.

**Comment by actual-nh** [comment]
I am not seeing a change to fields_test.cpp? (And I suggest that a weather override `with_windspeed` should be overriding _both_ weather _and_ windspeed in the same call (2 arguments); if need be, have a separate function doing just windspeed.)

**Comment by kevingranade** [comment]
After poking at the fire code, it looks like wind has a chance of blocking fire spread no matter what the intensity is, so I set wind direction as well, but it looks like this is probably not it as it still has a 50/50 chance of spreading despite wind if the strength is minimal.

File: tests/options_helpers.h [hunk 1]
```diff
@@ -26,6 +26,7 @@ class scoped_weather_override
     public:
         explicit scoped_weather_override( const weather_type_id & );
         void with_windspeed( const int windspeed_override );
+        void with_wind_direction( const int wind_directionspeed_override );
```
Comment (kevingranade) [inline tests/options_helpers.h:29]:
```suggestion
        void with_wind_direction( const int wind_direction_override );
```

**Comment by BrettDong** [comment]
Vision test keeps failing across multiple retries.

**Comment by kevingranade** [comment]
I'm completely at a loss for what's happening here, closing this out, it'll be in the history if I pick it up again before overhauling fire.

**Pull request closed without merging**